Path-Check/covid-safe-paths
# Improve status checks of background location service

#### Linked issues:

fixes issue 1 in #685 
when for the check for backgound service is still running, it was getting it wrong.  
and the fact that the variable isBackgroundGeolocationConfigured doesnt actually reflect the correct state of the BackgroundGeolocation in certain cases
also reinstates the code to open an Alert box allowing the user to open the settings and turn on location services.
also includes fix for issues #564 which was merged in

#### Screenshots:
![image](https://user-images.githubusercontent.com/22749929/80633634-fd4e3800-8a50-11ea-831b-1d9d8b2d9e4e.png)

#### How to test
for issue #685 part1
open app
background app ,
open settings and disable use location
go back to app.

Used to be - location tracking icon still green/ enabled
With fix will correctly show - red / disabled
App will detect this and pop up an Alert to allow the user to open the settings and turn on location services.

issue 564 see PR #574 for how to test

##### TODO 
localize resourse changes / additions
i think the settings app Alert opens twice sometimes.  Its doesnt break anything, its just a bit of a glitch
tests ... please advise!
